src/components/smart-house.js

```diff
@@ -0,0 +1,71 @@
+import Device from './device';
+
+class SmartHouse {
+    constructor(name) {
+      this._name = name;
+      this._devicesList = new Map();
+    }
+    get name() {
+      return this._name;
+    }
+    set name(name) {
+      if (typeof name === 'string' && name.length > 0) {
+        this._name = name;
+      }
+    }
+    
+    // I did insistent recommendation to create a function validating devices by name,
+    // but I suppose that the mentioned function is useless for my project, 
+    // because I use id-based system of identifying devices unlike name-based.
+
+    // _validateDeviceByName(name) {
+    //   const foundDevice = [...this._devicesList.values()].find((item) => {
+    //     return name.toLowerCase() === item.name.toLowerCase();
+    //   });
+    //   return foundDevice === undefined;
+    // }
+    addDevice(device) {
+      if (device instanceof Device /*&& this._validateDeviceByName(device.name)*/) {
+        const id = Math.floor(Math.random() * 100000000 + 1);
+        this._devicesList.set(id, device);
+      }
+    }
+    get devicesList() {
+      const list = [];
+
+      for (let item of this._devicesList.values()) {
+        list.push(item.name);
+      }
```

Comment:
`this._devicesList.values().map((item) => {
        list.push(item.name);
      });`
results in `Uncaught TypeError: this._devicesList.values(...).map is not a function` I think thats because values() return iterible collection and not an array, we could something like this:
`[...this._devicesList.values()].map((item) => {
        list.push(item.name);
      });`
to make it work, but I decided that it was redundant.